dbt_coves/tasks/generate/docs.py

```diff
@@ -0,0 +1,165 @@
+    pass
+
+
+class GenerateDocsTask(BaseConfiguredTask):
+    """
+    Task that generates content on local catalog.json
+    """
+
+    @classmethod
+    def register_parser(cls, sub_parsers, base_subparser):
+        subparser = sub_parsers.add_parser(
+            "docs",
+            parents=[base_subparser],
+            help="Merge models from a catalog.json into another one.",
+        )
+        subparser.add_argument(
+            "--merge-deferred",
+            action="store_true",
+            help="Flag to merge deferred models and sources into local catalog",
+            default=False,
+        )
+        subparser.add_argument(
+            "--state",
+            type=str,
+            help="Catalog.json to use as reference for merging",
+        )
+        cls.arg_parser = base_subparser
+        subparser.set_defaults(cls=cls, which="docs")
+        return subparser
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_config_value(self, key):
+        return self.coves_config.integrated["generate"]["docs"][key]
+
+    def _generate_dbt_docs(self):
+        output = run_and_capture_cwd(["dbt", "docs", "generate"], self.config.project_root)
+
+        if output.returncode == 0:
+            deps_status = "[green]SUCCESS :heavy_check_mark:[/green]"
+        else:
+            deps_status = "[red]FAIL :cross_mark:[/red]"
+        print_row(
+            "dbt docs generate",
+            deps_status,
+            new_section=True,
+        )
+        if output.returncode > 0:
+            raise Exception("dbt deps error. Check logs.")
+
+    def _fix_dbt_docs_links(self, docs_path: Path):
+        dbt_docs_index_path = Path(docs_path, "index.html")
+        new_tab_tag = "<base target='_blank'>"
+        with open(dbt_docs_index_path, "r") as f:
+            html_content = f.read()
+
+        head_start = html_content.find("<head>") + len("<head>")
+        head_end = html_content.find("</head>")
+
+        if new_tab_tag not in html_content[head_start:head_end]:
+            modified_content = html_content[:head_end] + new_tab_tag + html_content[head_end:]
+
+            # Write the modified content back to the HTML file
+            with open(dbt_docs_index_path, "w") as html_file:
+                html_file.write(modified_content)
+            console.print(
+                "[green]:heavy_check_mark:[/green] dbt docs updated. "
+                "External links will now open in a new tab"
+            )
+
+    def _get_catalog_json(self, docs_folder: Path) -> Dict[str, Any]:
+        """
+        Open json at docs_folder/catalog.json
+        """
+        catalog_path = Path(docs_folder, "catalog.json")
+        try:
+            with open(catalog_path, "r") as f:
+                return json.load(f)
+        except FileNotFoundError:
+            raise DbtCovesGenerateDocsException(f"Catalog.json not found at {catalog_path}")
+
+    def _merge_catalogs(
```

Comment:
why not using just `dict.update()`?

Reply:
Because we want to inform the user how many nodes/sources were merged.

I originally did it with `update` but I had to traverse dicts 2 times, one to update and the other to count.